### PR TITLE
feat(folding): support regions comments & eof & more

### PR DESCRIPTION
**Pull request by goulvenclech**
I added a lot of tests, and covered them.

### CODE REVIEW

File: src/features/folding.rs [hunk 1]
```diff
@@ -12,42 +22,164 @@ fn compute_folding_ranges(document: &TextDocumentItem) -> Vec<FoldingRange> {
         .chain(source.match_indices('\n').map(|(idx, _)| idx + 1))
         .collect();
```
Comment (Princesseuh) [inline src/features/folding.rs:23]: We have this already for the document in `StoreEntry#line_index.newlines`, this is quite expensive to calculate, so the one from the cache should always be used

File: src/features/folding.rs [hunk 2]
```diff
@@ -12,42 +22,164 @@ fn compute_folding_ranges(document: &TextDocumentItem) -> Vec<FoldingRange> {
...
             }
+            _ => {}
         }
     }
 
+    // Determine the last line with content
+    let mut total_lines = line_starts.len() as u32 - 1;
+    if source.ends_with('\n') && total_lines > 0 {
+        total_lines -= 1;
+    }
```
Comment (Princesseuh) [inline src/features/folding.rs:127]: What's the reason for wanting to return the last line with content vs the eof?

Comment (goulvenclech) [reply src/features/folding.rs:127]: Added a comment.

**Review by Princesseuh — APPROVED** [review]
Looks great, awesome work

**Pull request merged**